packages/gatsby/src/schema/data-tree-utils.js

```diff
@@ -14,24 +23,36 @@ const flatten = require(`flat`)
  *
  * @param {*Nodes} args
  */
-export const extractFieldExamples = (nodes: any[]) => {
+const extractFieldExamples = (nodes: any[]) => {
   // $FlowFixMe
-  return _.mergeWith({}, ...nodes, (obj, next) => {
-    if (!_.isArray(obj || next)) return
-    let array = [].concat(obj, next).filter(v => v != null)
+  return _.mergeWith({}, ...nodes, (obj, next, key, o, s, stack) => {
+    if (obj === INVALID_VALUE) return obj
+
+    // TODO: if you want to support infering Union types this should be handled
+    // differently. Maybe merge all like types into examples for each type?
+    // e.g. union: [1, { foo: true }, ['brown']] -> Union Int|Object|List
+    if (!isSameType(obj, next)) return INVALID_VALUE
```

Comment:
Hmmm ok. Can definitely see that union types make sense here. No need to figure out everything right now though. After a few more dozen source and transformer plugins the patterns will be a lot more clear. A bit busy today but will try to merge this later along with getting my rr V4 PR in.